tensorstore/driver/tiff/schema.yml

```diff
@@ -0,0 +1,166 @@
+  Provides read-only access to TIFF files or sequences of IFDs backed by a KeyValueStore.
+  Creation of new TIFF files is not supported.
+allOf:
+- $ref: KeyValueStoreBackedChunkDriver
+- type: object
+  properties:
+    driver:
+      const: tiff
+      description: Specifies the use of the read-only TIFF driver.
+    metadata:
+      title: TIFF dataset metadata constraints.
+      description: |
+        Specifies constraints that are validated against the metadata read
+        from the existing TIFF file. All members are optional. If specified,
+        they must match the corresponding properties inferred from the TIFF tags
+        and interpretation options.
+      allOf:
+      - type: object
+        properties:
+          dtype:
+            $ref: DataType
+            title: Data type constraint.
+            description: Constrains the expected data type of the TIFF dataset.
+          shape:
+            type: array
+            items:
+              type: integer
+              minimum: 0
+            title: Shape constraint.
+            description: |
+              Constrains the expected shape (dimensions) of the TIFF dataset. Length must match the rank.
+            examples:
+            - [ 1024, 1024, 50 ]
+    tiff:
+      title: TIFF-specific interpretation options.
+      description: Options controlling how the existing TIFF file or IFD sequence is interpreted.
+      type: object
+      properties:
+        ifd:
+          type: integer
+          minimum: 0
+          default: 0
+          title: Image File Directory (IFD) index.
+          description: |
+            Specifies which IFD to open when not using IFD stacking. Defaults to 0.
+            Cannot be non-zero if `ifd_stacking` is also specified.
+        ifd_stacking:
+          type: object
+          title: IFD Stacking Options.
+          description: |
+            Configures interpretation of a sequence of IFDs as additional TensorStore dimensions.
+            Cannot be specified if `ifd` is non-zero.
+          properties:
+            dimensions:
+              type: array
+              items:
+                type: string
+              minItems: 1
+              uniqueItems: true
+              title: Labels for stacked dimensions.
+              description: |
+                Required if `ifd_stacking` is used. Specifies the labels for the 
+                dimensions represented by the IFD sequence (e.g., ["z", "t"]).
+            dimension_sizes:
+              type: array
+              items:
+                type: integer
+                minimum: 1
+              title: Sizes of stacked dimensions.
+              description: |
+                Optional. Must match the length of `dimensions`. Required if `dimensions` 
+                has more than one entry. If only one dimension is specified, `ifd_count` can 
+                be used instead. If both are specified, their product must match `ifd_count`.
+            ifd_count:
```

Comment:
Why does this need to be specified --- can't we determine the number of IFDs from the file?

Reply:
I suppose it doesn't. I initially was thinking about wanting to use only a subset of IFDs, but then really `ifd_count` would be one of many parameters that a user would need to specify in order to provide them with the kind of flexibility to do that properly. I will remove.

Reply:
Actually, I double checked and I see now that IFDs may be ignored in OME XML so I will I need to add full specification functionality rather than remove this. https://docs.openmicroscopy.org/ome-model/5.6.3/ome-tiff/specification.html